File: VB/SampleMVVM/DataRowViewModel.vb
' Developer Express Code Central Example:
' How to handle a double-click on a grid row in a MVVM application
' 
' This example demonstrates one of many possible solutions for decoupling DXGrid
' mouse event handling from business logic, when using the MVVM architectural
' pattern.
' 
' You can find sample updates and versions for different programming languages here:
' http://www.devexpress.com/example=E2458
Imports System.Windows
Imports System.Windows.Input
Imports DevExpress.Xpf.Core.Commands

Namespace SampleMVVM

    Public Class DataRowViewModel

        Private _Row As DataRowModel, _CellDoubleClickCommand As ICommand

        Public Property Row As DataRowModel
            Get
                Return _Row
            End Get

            Private Set(ByVal value As DataRowModel)
                _Row = value
            End Set
        End Property

        Public Property CellDoubleClickCommand As ICommand
            Get
                Return _CellDoubleClickCommand
            End Get

            Private Set(ByVal value As ICommand)
                _CellDoubleClickCommand = value
            End Set
        End Property

        Public Sub New(ByVal row As DataRowModel)
            Me.Row = row
            CellDoubleClickCommand = New DelegateCommand(Of Object)(Sub(o) MessageBox.Show("Cell double click. Row: " & Me.Row.Id))
        End Sub
    End Class
End Namespace

File: CS/SampleMVVM/DataRowViewModel.cs
// Developer Express Code Central Example:
// How to handle a double-click on a grid row in a MVVM application
// 
// This example demonstrates one of many possible solutions for decoupling DXGrid
// mouse event handling from business logic, when using the MVVM architectural
// pattern.
// 
// You can find sample updates and versions for different programming languages here:
// http://www.devexpress.com/example=E2458

using System;
using System.Windows;
using System.Windows.Input;
using DevExpress.Xpf.Core.Commands;

namespace SampleMVVM {
    public class DataRowViewModel {
        public DataRowModel Row { get; private set; }
        public ICommand CellDoubleClickCommand { get; private set; }

        public DataRowViewModel(DataRowModel row) {
            Row = row;
            CellDoubleClickCommand = new DelegateCommand<object>(o => MessageBox.Show("Cell double click. Row: " + Row.Id));
        }
    }
}
